internal/argocd/argocdStructs.go

```diff
@@ -0,0 +1,255 @@
+package argocd
```

Comment:
idiomatic Go have types closer to their implementation, it's a bit bad to have to open another file to check the struct definition, I would like the idiomatic way

Reply:
Reading side by side is easier.. is go fan of 100s of line files?
Let me check k8s packages, I saw something different on some projects I checked.

Reply:
Do you have any reference for this style, searching for it.

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cronjob/cronjob_controllerv2.go

Reply:
nothing here: https://github.com/golovers/effective-go

Reply:
nothing that calls out here: https://go.dev/doc/effective_go#package-names

Reply:
> Reading side by side is easier.. is go fan of 100s of line files?
better a big file than trying to find what you need in different files. following that idea we soon will have
```
package-name/functions.go
package-name/types.go
package-name/methods.go
```

or
```
package-name/my-func.go
package-name/my-other-func.go
```

Reply:
maybe the day Kubefirst has a big chunk of types that are third party packages we can have it all on a single `types.go`, but it doesn't make much sense to do it for small packages. the argo package has one single file

Reply:
https://github.com/go-git/go-git/tree/master/internal/revision
For example: 
![image](https://user-images.githubusercontent.com/53096417/190717067-c44add4c-db2a-4e48-912c-bb6e92fd5158.png)

The idea was to isolate the code that is generated by yaml to struct from human code. 

On ideal design, as you said they would come from importing 3rd part pkg or isolating common data types on type packages k8s does. 

To avoid bloated code. 

As this is a candidate to PR conflict nightmare as size of file mix actions. I would agree to separate into smaller files by them, like application level action and argoCD global level actions, based on a semantic separation. 

But mix human/ tool generate code may bloat the code in a unnecessary way.

Reply:
Another sample: https://github.com/kubernetes/client-go/tree/master/applyconfigurations/core/v1

![image](https://user-images.githubusercontent.com/53096417/190717591-548333c7-9c02-42ef-b84d-80b24ca7cab3.png)

Reply:
Where to avoid a huge code base, and PR nightmare from multiple source an strategy of breaking smaller concerns were applied

Reply:
This struct are meant to expected to be updated as argo update its apis, the logic may persist for longer if the struct has only incremental changes.

Reply:
where does these examples have a file with structs only?

Reply:
Pkg that were broken into files to make easier to handle code that is growing.. I agree on having a definition of how to break it, but to farm huge files with huge structs is not a good practice.

Reply:
in this specific case (argo struct), I agree, the struct is huge, but there is a reason for that.

Argo API doesnt provide an endpoint to PATCH the application, for that we need to follow what Argo UI does that is firstly `GET` -> store it in the struct, and `PUT`.

I also a fan to have smaller functions and isolate, but there are better strategies for that.

Reply:
https://github.com/kubefirst/kubefirst/pull/410#issuecomment-1249977388